juju/juju
# Add firewaller support for provider role on consuming side of cmr

## Description of change

When the consuming side of a cross model relation has the provider role, different firewaller logic is needed. A new cross model watcher API (WatchEgressAddressesForRelation) is added to allow the firewaller worker on the consuming side to watch for egress address changes on the offer side.

Most of the diff in this PR is moving existing firewaller facade logic into a new common package so it can be shared between the firewaller and cross model facades.

Also includes a drive by fix for the output of the offer command.

## QA steps

Deploy prometheus and offer it.
Deploy telegraf in another model and relate.
Check that the telegraf model has firewall ports opened for port 9103 (telegraf port)
